src/Operations.jl

```diff
@@ -836,7 +828,7 @@ function _get_deps!(env::EnvCache, pkgs::Vector{PackageSpec}, uuids::Vector{UUID
 end
 
 function build(ctx::Context, pkgs::Vector{PackageSpec}, verbose::Bool)
-    if any_package_not_installed(ctx.env.manifest) || !isfile(ctx.env.manifest_file)
+    if !is_instantiated(ctx.env) || !isfile(ctx.env.manifest_file)
         Pkg.instantiate(ctx)
```

Comment:
There's a certain risk that `Pkg.instantiate` kicks of a build job itself here or? Anyway, the current change doesn't change that.